tests/test_fhir.py

```diff
@@ -0,0 +1,12 @@
+from medmorph_pha.models.process_message import process_message_operation
+
+
+def test_process_message(reporting_bundle_example, reporting_bundle_response_example):
+    response_bundle = process_message_operation(
+        reporting_bundle=reporting_bundle_example,
+        fhir_url="http://fake.com/fhir",
+    )
+
+    assert response_bundle["resourceType"] == "Bundle"
+    assert len(response_bundle["type"]) == len(reporting_bundle_response_example["type"])
+    assert len(response_bundle["entry"]) == len(reporting_bundle_response_example["entry"])
```

Comment:
Excellent...

Would you also want to use mocker to capture calls made to HAPI?  For whatever objects we expect this bundle and `$process_message` to generate, those HAPI API calls should get hit, right?  (can also be a later check-in if desired)

Reply:
I would have preferred to use mocker, but I am mocking (for my current understanding of mocking) the [`requests.post()` method here](https://github.com/uwcirg/medmorph-pha/pull/2/files#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R25-R40).

When data is POSTed to HAPI, HAPI just reflects the request body in the response, with the new logical ID assigned to the newly saved resource.

I may not be doing the traditional testing workflow of: mocking an object, then checking if a method has been called on that mocked object. However, doing it this way I can add assertions about the structure of the data returned by my custom "mock" object.

Let me know if I'm missing anything though- I certainly would prefer to use a mocking library for consistency but I think what I did still achieves similar results.

Reply:
I now understand, we're not there yet.  I was proposing you capture the calls to upsert the resources in the bundle - as they're sent to HAPI, by interrupting that flow via mock. But that will need to wait until we are persisting the objects :)